core/src/execution/datafusion/expressions/covariance.rs

```diff
@@ -287,22 +296,24 @@ impl Accumulator for CovarianceAccumulator {
     }
 
     fn evaluate(&mut self) -> Result<ScalarValue> {
+        if self.count == 0.0 {
+            return Ok(ScalarValue::Float64(None));
+        }
```

Comment:
Hmm, what it differs to previous? It previously returns null if `count` is 0 (divide by zero case). Now it is the same without checking `self.null_on_divide_by_zero`?

Reply:
The Spark implementation is:

CovPopulation:
```

if (n == 0.0) {
  return null
} else {
  return ck/n
}
```
CovSample:
```
if (n == 0.0) {
  return null
} else if (n==1.0) {
  if (nullOnDivideByZero) {
    return null
  } else { 
    return NaN
  }
} else {
  return ck/(n-1)
}

```

Reply:
Oh, the only difference is when `nullOnDivideByZero` is true, it could return null or NaN. Previously it is always null.